core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -65,6 +70,20 @@ class SnapshotService(protected val ctx: RawlsRequestContext,
       Future.successful(snapshotRef)
     }
 
+  // Ideally this would rely on Terra Policy Service, but until TPS is enabled for GCP
+  // We'll have to use this workaround for identifying protected status
+  def validateProtectedStatus(workspaceContext: Workspace, snapshot: NamedDataRepoSnapshot): Unit =
+    // logically it might make more sense to check if the snapshot is protected before the workspace
+    // but that is a more expensive check
+    // check if workspace is protected
+    if (!workspaceContext.bucketName.startsWith("fc-secure")) {
+      // if not, check if snapshot is protected
+      val sources = dataRepoDAO.getSnapshot(snapshot.snapshotId, ctx.userInfo.accessToken).getSource
+      if (sources.asScala.exists(_.getDataset.isSecureMonitoringEnabled)) {
```

Comment:
It's been a while since I've looked at scala but does this mean that a snapshot without any sources will not check this flag? Under what circumstances would a snapshot's sources be empty?

The only way I can see that sources would be missing is if the `include` param to the web API call didn't include `SOURCES`. Right now rawls is depending on the default API behavior to provide this field. It seems like the code should fail if that field is missing, since that means something has gone wrong.

https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/resources/api/data-repository-openapi.yaml#L642